packages/mattermost-redux/src/reducers/entities/emojis.ts

```diff
@@ -11,22 +12,27 @@ import {IDMappedObjects} from 'mattermost-redux/types/utilities';
 export function customEmoji(state: IDMappedObjects<CustomEmoji> = {}, action: GenericAction): IDMappedObjects<CustomEmoji> {
     switch (action.type) {
     case EmojiTypes.RECEIVED_CUSTOM_EMOJI: {
-        const nextState = {...state};
-        nextState[action.data.id] = action.data;
-        return nextState;
+        const emoji: CustomEmoji = action.data;
+
+        return storeEmoji(state, emoji);
     }
     case EmojiTypes.RECEIVED_CUSTOM_EMOJIS: {
-        const nextState = {...state};
-        for (const emoji of action.data) {
-            nextState[emoji.id] = emoji;
-        }
-        return nextState;
+        const emojis: CustomEmoji[] = action.data;
```

Comment:
same question here

Reply:
In this case, I think it's actually necessary so that TypeScript knows what the return value of `reduce` is. I might be confusing that with [this case below](https://github.com/mattermost/mattermost-webapp/pull/8552/files/8adc709f83665bf85b3c06bda3bdcbe3ae542a24#diff-205e6890da9eee5d067f7ded407255abd6fef79132519cdfbf2572733708dba4L42) that I also changed.